posts/crud-pattern.md

```diff
@@ -0,0 +1,294 @@
+.. title: CRUD Pattern
+.. slug: crud-pattern
+.. date: 2020-03-12 17:02:28 UTC+05:30
+.. tags:
+.. category:
+.. link:
+.. description:
+.. status: draft
+.. type: text
+
+# CRUD Pattern
+
+React.js based frontend powered by REST API can be a typical architecture in Dashboard like applications. In my last project, I was working on such a Dashboard application where front-end was written in React.js and the API was powered by Django REST framework. The application consisted of multiple pages with a similar structure. Each page would look like a table of records which can be **Created**, **Retrieved**, **Updated** or **Deleted**, the common **CRUD** pattern. When I joined the project, the prototype was already in place and the code had become complex to understand. After several sessions of work on different parts of the codebase, it was apparent that each page had similar functionality(CRUD actions). There seemed a possibility of abstraction of this functionality into a common component that would provide a neatly structured code that is easier to understand and maintain.
```

Comment:
Let's not use the word 'common component'. It doesn't convey the same thing as 'pattern'. Also, this para should be shortened by about 30%. Let's think about it.